### PR TITLE
Add documentation for internal architecture.

### PR DESCRIPTION
**Pull request by hodgestar**
We have a good explanation of how Junebug looks from the outside, but it would help to have a description of how it looks form the inside.

### CODE REVIEW

**Comment by hodgestar** [comment]
selection_067

**Comment by hodgestar** [comment]
@rudigiesler @bruskiza Ready for review.

File: docs/internal-architecture.rst [hunk 1]
```diff
@@ -0,0 +1,84 @@
+Internal Architecture
+=====================
+
+Internall Junebug is structured as a set of services that live within a
```
Comment (rudigiesler) [inline docs/internal-architecture.rst]: s/Internall/Internal

**Comment by rudigiesler** [comment]
Ready for review.

**Comment by smn** [comment]
:+1:

**Pull request merged**